tests/nix_tests/utils.py

```diff
@@ -264,8 +381,116 @@ def compare_fields(a, b, fields):
 # get_fees_from_tx_result returns the fees by unpacking them
 # from the events contained in the tx_result of a cosmos transaction.
 def get_fees_from_tx_result(tx_result, denom=DEFAULT_DENOM):
-    for event in tx_result["events"]:
-        if event["type"] == "tx":
-            for attr in event["attributes"]:
-                if attr["key"] == "fee":
-                    return int(attr["value"].split(denom)[0])
+    return int(
+        get_event_attribute_value(
+            tx_result["events"],
+            "tx",
+            "fee",
+        ).split(
+            denom
+        )[0]
+    )
+
+
+def get_event_attribute_value(events, _type, attribute):
+    for event in events:
+        if event["type"] == _type:
+            attrs = event["attributes"]
+            for attr in attrs:
+                if attr["key"] == attribute:
+                    return attr["value"]
+
+
+def update_node_cmd(path, cmd, i):
+    ini_path = path / SUPERVISOR_CONFIG_FILE
+    ini = configparser.RawConfigParser()
+    ini.read(ini_path)
+    for section in ini.sections():
+        if section == f"program:evmos_9000-1-node{i}":
+            ini[section].update(
+                {
+                    "command": f"{cmd} start --home %(here)s/node{i}",
+                    "autorestart": "false",  # don't restart when stopped
+                }
+            )
+    with ini_path.open("w") as fp:
+        ini.write(fp)
+
+
+def update_evmos_bin(modified_bin, nodes=[0, 1]):
+    """
+    updates the evmos binary with a patched binary.
+    Input parameters are the modified binary (modified_bin)
+    and the nodes in which
+    to apply the modified binary (nodes).
+    Usually the setup comprise only 2 nodes (node0 & node1),
+    so nodes should be an array containing only 0 and/or 1
+    """
+
+    def inner(path, base_port, config):
+        chain_id = "evmos_9000-1"
+        # by default, there're 2 nodes
+        # need to update the bin in all these
+        for i in nodes:
+            update_node_cmd(path / chain_id, modified_bin, i)
+
+    return inner
+
+
+def register_wevmos(evmos):
+    """
+    this helper function registers the WEVMOS
+    token in the ERC20 module and returns the contract address.
+    Make sure to patch the evmosd binary with the allow-wevmos-register patch
+    for this to be successful
+    """
+    cli = evmos.cosmos_cli()
+    proposal = {
+        "title": "Register WEVMOS",
+        "description": "EVMOS erc20 representation",
+        "metadata": [WEVMOS_META],
+        "deposit": "1aevmos",
+    }
+    proposal_id = register_ibc_coin(cli, proposal)
+    assert int(proposal_id) > 0
```

Comment:
nit: some more verbose error messages wouldn't hurt IMO :) 
```suggestion
    assert int(proposal_id) > 0, "expected a non-zero proposal ID for the registration of the WEVMOS token."
```